### PR TITLE
RocksDB integration

### PR DESCRIPTION
**Pull request by LucioDonda**
|Related issue|
|---|
|#12617|

## Description

Added RockDB integration
added simple example 

## Tests

added test calling simple example

### CODE REVIEW

File: src/engine/CMakeLists.txt [hunk 1]
```diff
@@ -102,6 +102,8 @@ add_subdirectory(${ENGINE_SOURCE_DIR}/shared)
 add_subdirectory(${ENGINE_SOURCE_DIR}/catalog)
 add_subdirectory(${ENGINE_SOURCE_DIR}/server)
 add_subdirectory(${ENGINE_SOURCE_DIR}/router)
+add_subdirectory(${ENGINE_SOURCE_DIR}/kvdb)
+# TODO dependent
```
Comment (palaciosjeremias) [inline src/engine/CMakeLists.txt:106]: This comment was eliminated on previous Engine commits, and we should keep it deleted.

File: src/engine/source/kvdb/include/kvdb/kvdb.hpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#ifndef _KVDB_H
+#define _KVDB_H
+#include <functional>
```
Comment (palaciosjeremias) [inline src/engine/source/kvdb/include/kvdb/kvdb.hpp:3]: We should clean this include until they are really needed.

File: src/engine/source/kvdb/src/kvdb.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
+#include <algorithm>
+#include <functional>
+#include <string>
+#include <string_view>
```
Comment (palaciosjeremias) [inline src/engine/source/kvdb/src/kvdb.cpp:4]: Ditto

File: src/engine/test/source/kvdb/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    GITHUB_REPOSITORY Tencent/rapidjson
+    GIT_TAG master
+    VERSION 1.1.0
+    OPTIONS
+        "RAPIDJSON_BUILD_DOC OFF"
+        "RAPIDJSON_BUILD_EXAMPLES OFF"
+        "RAPIDJSON_BUILD_TESTS OFF"
+)
+
+# Configuration Catalog tests
```
Comment (palaciosjeremias) [inline src/engine/test/source/kvdb/CMakeLists.txt:13]: We should rename this comment.

**Review by palaciosjeremias — CHANGES_REQUESTED** [review]

File: src/engine/source/kvdb/src/kvdb.cpp (hunk 3 above)
Comment (LucioDonda) [reply src/engine/source/kvdb/src/kvdb.cpp]: done

File: src/engine/source/kvdb/include/kvdb/kvdb.hpp (hunk 2 above)
Comment (LucioDonda) [reply src/engine/source/kvdb/include/kvdb/kvdb.hpp]: done

File: src/engine/source/kvdb/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+# Sources - Includes
+####################################################################################################
+add_library(kvdb ${ENGINE_KVDB_SOURCE_DIR}/kvdb.cpp)
+
+target_link_libraries(kvdb PRIVATE rocksdb)
+
+target_include_directories(kvdb PUBLIC
+    ${ENGINE_KVDB_INCLUDE_DIR}
+    ${CMAKE_BINARY_DIR}/_deps/rocksdb-src/include
+)
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/source/kvdb/CMakeLists.txt]: done

File: src/engine/source/kvdb/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+CPMAddPackage(
+  NAME rocksdb
+  GITHUB_REPOSITORY facebook/rocksdb
+  GIT_TAG v6.29.3
+  VERSION 6.29.3
+  OPTIONS
+    "WITH_GFLAGS OFF"
+    "WITH_TESTS OFF"
+    "WITH_BENCHMARK_TOOLS OFF" # this flag must be off to avoid test colition
```
[the remark this answers is not in the document]
Comment (LucioDonda) [reply src/engine/source/kvdb/CMakeLists.txt]: done

**Review by palaciosjeremias — APPROVED** [review]
LGTM

**Pull request merged**